### PR TITLE
fix: [Type] useAccount.address should return an Address type

### LINKED ISSUE #452: [Type] useAccount.address should return an Address type
`useAccount()` hook return the following type: 

```
/** Value returned from `useAccount`. */
export type UseAccountResult = {
  /** The connected account object. */
  account?: AccountInterface;
  /** The address of the connected account. */
  address?: string;
  /** The connected connector. */
  connector?: Connector;
  /** Connector's chain id */
  chainId?: bigint;
  /** True if connecting. */
  isConnecting?: boolean;
  /** True if reconnecting. */
  isReconnecting?: boolean;
  /** True if connected. */
  isConnected?: boolean;
  /** True if disconnected. */
  isDisconnected?: boolean;
  /** The connection status. */
  status: AccountStatus;
};
```

The `address` should be typed as `Address` instead of `string`

### PR DESCRIPTION
**Pull request by Benjtalkshow**
Closes #452 

To address the issue with address formatting in the `useAccount` hook, I have implemented a new `getAddress` function. This function ensures that addresses are consistently prefixed with 0x, aligning with the expected Address type.

### CODE REVIEW

File: pnpm-lock.yaml
Comment (fracek) [inline pnpm-lock.yaml:1]: Can you remove this file from the changes (NOTE: don't remove the file, keep the version that was already there).

File: packages/chains/tsup.config.bundled_cmy92unsufi.mjs
Comment (fracek) [inline packages/chains/tsup.config.bundled_cmy92unsufi.mjs:1]: This file was included by accident. Remove it please.

File: packages/core/src/hooks/use-account.ts [hunk 1]
```diff
@@ -37,6 +35,19 @@ export type UseAccountResult = {
...
 };
 
+/** 
+ * Validate and format the address.
+ *
+ * @param address - The address string to validate.
+ * @returns The validated and formatted address as Address type.
+ */
+function getAddress(address: string): Address {
+  return address.startsWith("0x")
```
Comment (fracek) [inline packages/core/src/hooks/use-account.ts:45]: Can you use the [`validateAndParseAddress`](https://github.com/starknet-io/starknet.js/blob/develop/src/utils/address.ts#L41) function from starknet.js? Also move this function to a file `utils.ts` in `packages/core/src` and then export it from `packages/core/src/index.ts`

File: pnpm-lock.yaml
Comment (Benjtalkshow) [reply pnpm-lock.yaml:1]: @fracek 
Did you mean I should revert the version that was there before but I shouldn't delete the file entirely?

**Comment by Benjtalkshow** [comment]
@fracek i have updated my PR

File: packages/chains/tsup.config.bundled_cmy92unsufi.mjs
Comment (fracek) [inline packages/chains/tsup.config.bundled_cmy92unsufi.mjs:1]: Please, make sure to never commit these files!

File: packages/core/src/hooks/use-account.ts [hunk 2]
```diff
@@ -37,6 +37,14 @@ export type UseAccountResult = {
...
 };
 
+/** 
+ * Validate and format the address.
+ *
+ * @param address - The address string to validate.
+ * @returns The validated and formatted address as Address type.
+ */
+
+
```
Comment (fracek) [inline packages/core/src/hooks/use-account.ts:47]: The docstring must go with the function.

File: packages/core/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Address } from "@starknet-react/chains";
+import { validateAndParseAddress } from "starknet";
+
+export function getAddress(address: string): Address {
+    const validatedAddress = validateAndParseAddress(address);
+    return validatedAddress as Address;
```
Comment (fracek) [inline packages/core/src/utils.ts:6]:
```suggestion
    return validateAndParseAddress(address) as Address;
```

**Comment by fracek** [comment]
I added a couple more comments. Also run `pnpm lint:fix`

File: packages/core/src/hooks/use-account.ts (hunk 2 above)
Comment (Benjtalkshow) [reply packages/core/src/hooks/use-account.ts:47]: > The docstring must go with the function.

is this the only changes to make?

**Comment by Benjtalkshow** [comment]
@fracek 
I have placed the `docString` close to the function and simplified the code. I also ran `pnpm fix:lint`

**Comment by Benjtalkshow** [comment]
@fracek 
I have placed the `docString` close to the function and simplified the code.
I also ran `pnpm fix:lint`

**Comment by Benjtalkshow** [comment]
@fracek 
I have placed the docString` close to the function and simplified the code.
I also ran `pnpm fix:lint`

File: pnpm-lock.yaml
Comment (fracek) [reply pnpm-lock.yaml:1]: Basically, make sure the version used to generate the lockfile is pnpm 8 or 9.

**Comment by Benjtalkshow** [comment]
> Looks good now! Once you fix formatting and the CI passes I will merge it.

@fracek Please i updated the repo

**Comment by Benjtalkshow** [comment]
@fracek Please i updated the repo

**Comment by Benjtalkshow** [comment]
> Looks like the new code caught an issue in the old code! Please fix the test and make sure linting and tests work before pushing.

Sir it seems each time I run lint, it adds another issue. How can I avoid this?

**Comment by fracek** [comment]
Run `lint:fix` and `test` locally and fix issues as they come.

**Comment by fracek** [comment]
Thank you for this!

**Comment by Benjtalkshow** [comment]
> Thank you for this!

You're welcome sir

**Pull request merged**